diff --git a/accel-pppd/ctrl/ipoe/ipoe.c b/accel-pppd/ctrl/ipoe/ipoe.c
index 5438b61..5a8d40a 100644
--- a/accel-pppd/ctrl/ipoe/ipoe.c
+++ b/accel-pppd/ctrl/ipoe/ipoe.c
@@ -726,7 +726,7 @@ static void find_gw_addr(struct ipoe_session *ses)
 	list_for_each_entry(a, &conf_gw_addr, entry) {
 		if ((ntohl(ses->yiaddr) & (a->mask1)) == (ntohl(a->addr) & (a->mask1))) {
 			ses->router = a->addr;
-			if (!ses->mask)
+			if (!ses->mask || ses->mask == 32)
 				ses->mask = a->mask;
 			return;
 		}
